Makefile

```diff
@@ -13,7 +16,11 @@ test-clj: .source-deps
 	lein with-profile +$(VERSION),+test-clj test
 
 test-cljs: .source-deps
+ifeq ($(JAVA_VERSION),:java9)
+	lein with-profile +$(VERSION),+test-cljs,+jxb test
+else
 	lein with-profile +$(VERSION),+test-cljs test
+endif
 
```

Comment:
Oh well. Let's go ahead with the Makefile version then. Does this look ok ? I couldn't figure out how to use both `with-profile` and `update-in` at the same time. I'm not using Clojure/Leiningen for a while now. And I ran out of make-fu :P . Could you please help with this ?

Reply:
Does that work? I've never tried expressing a recipe that way. I would have done the conditional test in bash syntax, ie. all the logic completely inside the recipe.

```
test-cljs: .source-deps
        if [ "$(JAVA_VERSION)" = "9.0" ]; then \
            lein with-profile +$(VERSION),+test-cljs,+jxb test; \
        else \
            lein with-profile +$(VERSION),+test-cljs test; \
        fi
```

Also, now that I'm thinking about it...   I wonder if it's more future-proof to name the Leiningen profile "+java9.0" rather than "+jxb"?  In the event the project requires other Java 9 modules down the road then this mechanism for adding modules has a general label instead of the specific one.

Reply:
while I'm at my keyboard I'll take a look at the `update-in` usage also.

Reply:
Doing it with `update-in` would look something like this:
```
test-cljs: .source-deps
        if [ "$(JAVA_VERSION)" = "9.0" ]; then \
            lein with-profile +$(VERSION),+test-cljs \
                 update-in :jvm-opts concat "[\"--add-module\" \"java.xml.bind\"]" \ 
                 -- test; \
        else \
            lein with-profile +$(VERSION),+test-cljs test; \
        fi
```

going this route you don't need the `:jxb` profile in `project.clj` at all.